src/main/java/com/minecolonies/coremod/entity/ai/citizen/netherworker/EntityAIWorkNether.java

```diff
@@ -0,0 +1,958 @@
+
+    @Override
+    public boolean canBeInterrupted()
+    {
+        return !worker.isInvisible();
+    }
+
+    @Override
+    protected IAIState decide()
+    {
+        if(job.isInNether())
+        {
+            if (!worker.isInvisible())
+            {
+                worker.setInvisible(true);
+            }
+            return NETHER_AWAY;
+        }
+        if (worker.isInvisible())
+        {
+            worker.setInvisible(false);
+        }
+
+        IAIState crafterState = super.decide();
+
+        if(crafterState != IDLE && crafterState != START_WORKING)
+        {
+            return crafterState;
+        }
+
+        // Get Armor if available. 
+        // This is async, so we could go to the nether without it. 
+        checkAndRequestArmor();
+
+        // Check for materials needed to go to the Nether: 
+        IRecipeStorage rs = getOwnBuilding().getFirstModuleOccurance(BuildingNetherWorker.CraftingModule.class).getFirstRecipe(ItemStack::isEmpty);
+        if(rs != null)
+        {
+            for(ItemStorage item : rs.getInput())
+            {
+                checkAndRequestMaterials(item.getItemStack(), item.getAmount());
+            }
+        }
+
+        // Make sure we have a stash of some food 
+        checkAndRequestFood(16);
+
+        // Get other adventuring supplies. These are required. 
+        // Done this way to get all the requests in parallel
+        boolean haveAxe = checkForToolOrWeapon(ToolType.AXE);
+        boolean havePick = checkForToolOrWeapon(ToolType.PICKAXE);
+        boolean haveShovel = checkForToolOrWeapon(ToolType.SHOVEL);
+        boolean haveSword = checkForToolOrWeapon(ToolType.SWORD);
+        boolean haveLighter = checkForToolOrWeapon(ToolType.FLINT_N_STEEL);
+        if(haveAxe || havePick || haveShovel || haveSword || haveLighter)
+        {
+            worker.getCitizenData().setIdleAtJob(true);
+            return IDLE;
+        }
+
+        if (currentRecipeStorage == null)
+        {
+            final ICraftingBuildingModule module = getOwnBuilding().getFirstModuleOccurance(BuildingNetherWorker.CraftingModule.class);
+            currentRecipeStorage = module.getFirstFulfillableRecipe(ItemStackUtils::isEmpty, 1, false);
+            if(getOwnBuilding().canDoTrip())
+            {
+                worker.getCitizenData().setIdleAtJob(true);
+            }
+
+            final BlockPos portal = getOwnBuilding().getPortalLocation();
+            if(portal != null && currentRecipeStorage == null && getOwnBuilding().shallClosePortalOnReturn())
+            {
+                final BlockState block = world.getBlockState(portal);
+                if (block.is(Blocks.NETHER_PORTAL))
+                {
+                    return NETHER_CLOSEPORTAL;
+                }
+            }
+
+            return getState();
+        }
+        else 
+        {
+            if (!getOwnBuilding().canDoTrip())
+            {
+                worker.getCitizenData().setIdleAtJob(false);
```

Comment:
Isn't that idle true then?

Reply:
No, this is the case when we *can't* do the job, potentially for days, because we've done one too recently. That's not idle, that's recovering. Otherwise we hit the idle state way too fast.

Reply:
so setIdleAtJob = true -> we're not idle and false -> we're idle?

Reply:
That's not what I said. when !canDoTrip, we're not idle, because we're recovering from our last trip. If we set period to a multi day period, say 3 days, and our idle timeout before the gear is 3 days, then we're at high risk of the gear, even though we can't do the job because we're counting down the timeout. This way, we're not "idle" until we could go to the nether and haven't.

Reply:
Ahhhh, understood. Maybe we rename canDoTrip then to sth "isTakingABreak" or so.

Reply:
Let me think about that. How about "noRecentTrip"? So !noRecentTrip is logically consistent with canDoTrip, but reads better.

Reply:
Hmm.. I don't know that I like that either. It's structurally designed to support 2 trips every 3 days.

Reply:
isReadyForTrip that implies more "waiting" rather than "lack of sth"

Reply:
Done and done.